File: SAS/lib/departments.ts
// =============================================================
//  lib/departments.ts
//
//  A super user runs exactly one freight desk: air or sea. The desk lives in
//  profiles.department, but is not stored consistently — the same desk appears
//  as 'sea' and 'Sea'. Compare through freightMode() rather than the raw string.
//
//  Mirrors Backend/utils/departments.py — keep the two in step.
// =============================================================

export type FreightMode = 'AIR' | 'SEA';

const AIR_SPELLINGS = ['air', 'airfreight', 'air freight'];
const SEA_SPELLINGS = ['sea', 'seafreight', 'sea freight', 'ocean'];

/** 'Sea' -> 'SEA'. Returns null when the department is not a freight desk. */
export function freightMode(department: string | null | undefined): FreightMode | null {
  const key = (department ?? '').trim().toLowerCase();
  if (AIR_SPELLINGS.includes(key)) return 'AIR';
  if (SEA_SPELLINGS.includes(key)) return 'SEA';
  return null;
}

/** Freight desk of the signed-in user, from what the login stored. */
export function storedFreightMode(): FreightMode | null {
  if (typeof window === 'undefined') return null;
  return freightMode(localStorage.getItem('user_department'));
}

File: Backend/utils/departments.py
"""
departments.py — freight desk helpers.

A super user runs exactly one freight desk: air or sea. That desk is stored in
profiles.department, but not consistently — the same desk appears as 'sea',
'Sea' and (for air) 'air'. Compare through normalize_department() rather than
against the raw string, otherwise 'Sea' and 'sea' look like different desks.
"""

AIR = 'AIR'
SEA = 'SEA'

_AIR_SPELLINGS = {'air', 'airfreight', 'air freight'}
_SEA_SPELLINGS = {'sea', 'seafreight', 'sea freight', 'ocean'}


def normalize_department(value):
    """
    Map a stored department to 'AIR' / 'SEA', or None when it is not a freight
    desk (empty, 'Sales', 'operations', ...).
    """
    key = (value or '').strip().lower()
    if key in _AIR_SPELLINGS:
        return AIR
    if key in _SEA_SPELLINGS:
        return SEA
    return None


def is_super_user(role):
    """True for any stored spelling of the super user role ('Super User', 'superuser')."""
    return (role or '').replace(' ', '').replace('_', '').replace('-', '').lower() == 'superuser'
